### PR TITLE
Add Clone to MaybeUnset and SessionBuilder

### PR DESCRIPTION
**Pull request by Gor027**
## Pre-review checklist

- [x] I have split my patch into logically separate commits.
- [x] All commit messages clearly explain what they change and why.
- [ ] I added relevant tests for new features and bug fixes.
- [x] All commits compile, pass static checks and pass test.
- [x] PR description sums up the changes and reasons why they should be introduced.
- [ ] I added appropriate `Fixes:` annotations to PR description.

Added Clone and Copy trait implementations for MaybeUnset enum as it is
necessary in cpp-rust-driver to copy CassStatement values.

Added Clone trait implementation to SessionBuilder in order to clone
cluster configuration while creating a session object.

### CODE REVIEW

File: scylla/src/frame/value.rs [hunk 1]
```diff
@@ -32,6 +32,7 @@ pub struct Unset;
 pub struct Counter(pub i64);
 
 /// Enum providing a way to represent a value that might be unset
+#[derive(Clone, Copy)]
```
Comment (psarna) [inline scylla/src/frame/value.rs:35]: contrary to to the commit title, it also adds Copy here, not just Clone. I don't think it's desired, not all Value impls are Copy. I'm actually surprised it compiles - is there some rule in Rust I'm not aware of that would only allow Copy semantics here is V is also Copy?

**Review by psarna — COMMENTED** [review]
The commit misses its commit message - please add a few words, e.g. what's the motivation for this change.

File: scylla/src/frame/value.rs (hunk 1 above)
Comment (piodul) [reply scylla/src/frame/value.rs:35]: Yes, it works like that. If you add `#[derive(Clone)]` or Copy for a generic struct, it only works when the generic type parameters implement Clone/Copy, too.

Comment (psarna) [reply scylla/src/frame/value.rs:35]: Ok, it's good then. So the only thing to fix is to mention Copy in the commit message as well.

Comment (piodul) [reply scylla/src/frame/value.rs:35]: Some source for the above claim: https://doc.rust-lang.org/std/clone/trait.Clone.html#derivable

I'd expect it's similar for Copy.

**Review by piodul — APPROVED** [review]

**Pull request merged**